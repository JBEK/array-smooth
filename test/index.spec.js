/* global describe, it, expect */

const arraySmooth = require('../lib')

describe('arraySmooth', () => {
  it('expose a function as default export', () => {
    expect(typeof arraySmooth).toBe('function')
  })
  it('smooths a given array', () => {
    const arr = [1,2,3,4,5,6,7,8,9,10]
    const range = 2
    expect(arraySmooth(arr, range)).toMatchSnapshot()
  })
  it('smooths a given array using a getter', () => {
    const arr = [1,2,3,4,5,6,7,8,9,10]
    const getter = (obj) => obj.value
    const range = 2
    expect(arraySmooth(arr.map((number) => ({value: number}) ), range, getter)).toMatchSnapshot()
  })
  it('smooths a given array using a getter', () => {
    const arr = [1,2,3,4,5,6,7,8,9,10]
    const getter = (obj) => obj.value
    const range = 2
    expect(arraySmooth(arr.map((number) => ({value: number}) ), range, getter)).toMatchSnapshot()
  })
})