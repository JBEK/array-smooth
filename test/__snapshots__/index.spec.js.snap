// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arraySmooth smooths a given array 1`] = `
Array [
  2,
  2.5,
  3,
  4,
  5,
  6,
  7,
  8,
  8.5,
  9,
]
`;

exports[`arraySmooth smooths a given array using a getter function 1`] = `
Array [
  3,
  3.5,
  4,
  4.5,
  5,
  6,
  6.5,
  7,
  7.5,
  8,
]
`;

exports[`arraySmooth smooths a given array using a getter string 1`] = `
Array [
  1.5,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  9.5,
]
`;

exports[`arraySmooth smooths a given array using a setter function 1`] = `
Array [
  Object {
    "smoothed": Object {
      "value": 2,
    },
    "value": 1,
  },
  Object {
    "smoothed": Object {
      "value": 2.5,
    },
    "value": 2,
  },
  Object {
    "smoothed": Object {
      "value": 3,
    },
    "value": 3,
  },
  Object {
    "smoothed": Object {
      "value": 4,
    },
    "value": 4,
  },
  Object {
    "smoothed": Object {
      "value": 5,
    },
    "value": 5,
  },
  Object {
    "smoothed": Object {
      "value": 6,
    },
    "value": 6,
  },
  Object {
    "smoothed": Object {
      "value": 7,
    },
    "value": 7,
  },
  Object {
    "smoothed": Object {
      "value": 8,
    },
    "value": 8,
  },
  Object {
    "smoothed": Object {
      "value": 8.5,
    },
    "value": 9,
  },
  Object {
    "smoothed": Object {
      "value": 9,
    },
    "value": 10,
  },
]
`;

exports[`arraySmooth smooths a given array using a setter string 1`] = `
Array [
  Object {
    "s": Object {
      "item": Object {
        "smoothed": 2,
      },
      "value": 1,
    },
  },
  Object {
    "s": Object {
      "item": Object {
        "smoothed": 2.5,
      },
      "value": 2,
    },
  },
  Object {
    "s": Object {
      "item": Object {
        "smoothed": 3,
      },
      "value": 3,
    },
  },
  Object {
    "s": Object {
      "item": Object {
        "smoothed": 4,
      },
      "value": 4,
    },
  },
  Object {
    "s": Object {
      "item": Object {
        "smoothed": 5,
      },
      "value": 5,
    },
  },
  Object {
    "s": Object {
      "item": Object {
        "smoothed": 6,
      },
      "value": 6,
    },
  },
  Object {
    "s": Object {
      "item": Object {
        "smoothed": 7,
      },
      "value": 7,
    },
  },
  Object {
    "s": Object {
      "item": Object {
        "smoothed": 8,
      },
      "value": 8,
    },
  },
  Object {
    "s": Object {
      "item": Object {
        "smoothed": 8.5,
      },
      "value": 9,
    },
  },
  Object {
    "s": Object {
      "item": Object {
        "smoothed": 9,
      },
      "value": 10,
    },
  },
]
`;
